### DISCUSSION TITLE
process.env.NODE_ENV doesnt work in .eslint.js when using vite

### DISCUSSION THREAD

**Discussion by Mr-Nobody-li** [eslint/eslint#16528] [Q&A]
I don't know whether this is a bug or whether there are other ways to judge the environment for vite
I found this cannot work because i cannot get `process.env.NODE_ENV`
```
rules:{
  'no-debugger': process.env.NODE_ENV === 'development' ? 'off' : 'error'
}
 ```
I was wondering whether it was unnecessary to write like this, after all, the production environment could not use eslinlt

**Comment by harish-sethuraman** [comment]
This is likely not a problem with ESLint and also on a note if `the production environment could not use eslinlt` then there is no point in having the rule for production environment?

**Reply by Mr-Nobody-li** [reply]
Yes, later I realized that there is no eslint in the production environment, so now I write it like this
```
rules:{
  'no-debugger': 'off'
}
```
But I still don't know why `process.env` cannot be output in `vite`, while `webpack` can. Maybe I should start discussions in vite repo

**Reply by harish-sethuraman** [reply]
May be this has what you are looking for https://github.com/vitejs/vite/issues/1930

**Comment by Fideltodayy** [comment]
Did you get an answer to this?
I am stranded with the same but I think its how vite is configured
